docs/src/dynamichmc.md

```diff
@@ -0,0 +1,27 @@
+---
+title: Using DynamicHMC
+permalink: /docs/dynamichmc/
+---
+
+Turing supports the use of [DynamicHMC](https://github.com/tpapp/DynamicHMC.jl) as a sampler through the use of the `DynamicNUTS` function. This is a [faster](https://github.com/TuringLang/Turing.jl/issues/559) version of Turing's native `NUTS` implementation.
```

Comment:
Thanks, @cpfiffer. This is quite helpful. Perhaps we should give some (possible) explanations on why Turing.NUTS is slow compared to DynamicHMC:

- Turing.NUTS adapt pre-conditioning matrix

@xukai92 any other notable differences between Turing.NUTS and DynamicHMC?

and also say `Turing.NUTS` support compositional inference (i.e. can be used together within a Gibbs sampler) and its efficiency will slowly catch up with other implementations.